src/audio/src/src_generic.c

```diff
@@ -101,16 +96,16 @@ static inline void fir_filter_generic(int32_t *rp, const void *cp, int32_t *wp0,
 		 * Qx.31.
 		 */
 		for (i = 0; i < n1; i += nch) {
-			y0 += (int64_t)(*coef) * (*data);
-			coef++;
+			y0 += (int64_t)(*(coef++)) * (*data);
```

Comment:
for (i = 0; i < n1; i += nch, data += nch) 
			y0 += (int64_t)(*(coef++)) * (*data);

How about this format? more compact?

Reply:
As this is small optimization, may not improve performance(mcps) directly, we can try to make code looks better, above change can remove one pair {} , worth to do.